lib/Controller/SettingsController.php

```diff
@@ -92,8 +92,13 @@ public function updateAppConfig(string $settingsKey, $settingsValue): JSONRespon
 			}
 		}
 
-		// Set on DB
-		$this->config->setAppValue($this->appName, $settingsKey, json_encode($settingsValue));
+		if ($settingsKey === Constants::CONFIG_KEY_LOGLEVEL) {
+			// Set backend loglevel directly via system value
+			$this->config->setSystemValue('loglevel', $settingsValue);
```

Comment:
I would like to have some validation here, as you get a probably unknown user input and set it directly into the server config. So something like:

```suggestion
			if (!is_int($settingsValue) || $settingsValue < 0 || $settingsValue > 4) {
				return new JSONResponse(statusCode: Http::STATUS_BAD_REQUEST);
			}
			$this->config->setSystemValue('loglevel', $settingsValue);
```